src/query/expression/src/deserializations/tuple.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_io::prelude::FormatSettings;
+
+use crate::types::DataType;
+use crate::Column;
+use crate::Scalar;
+use crate::TypeDeserializer;
+
+pub struct TupleDeserializer {
+    pub inners: Vec<Box<dyn TypeDeserializer>>,
+}
+
+impl TupleDeserializer {
+    pub fn with_capacity(capacity: usize, inners: &[DataType]) -> Self {
+        let inners = inners
+            .iter()
+            .map(|d| d.create_deserializer(capacity))
+            .collect();
+        Self { inners }
+    }
+}
+
+impl TypeDeserializer for TupleDeserializer {
+    fn memory_size(&self) -> usize {
+        self.inners.iter().map(|d| d.memory_size()).sum()
+    }
+
+    fn de_binary(&mut self, reader: &mut &[u8], format: &FormatSettings) -> Result<()> {
+        for inner in self.inners.iter_mut() {
+            inner.de_binary(reader, format)?;
+        }
+        Ok(())
+    }
+
+    fn de_default(&mut self) {
+        for inner in self.inners.iter_mut() {
+            inner.de_default();
+        }
+    }
+
+    fn de_json(&mut self, value: &serde_json::Value, format: &FormatSettings) -> Result<()> {
+        match value {
+            serde_json::Value::Object(obj) => {
+                if self.inners.len() != obj.len() {
+                    return Err(ErrorCode::from_string(format!(
+                        "Incorrect json value, expect {} values, but get {} values",
+                        self.inners.len(),
+                        obj.len()
+                    )));
+                }
+                for (inner, item) in self.inners.iter_mut().zip(obj.iter()) {
```

Comment:
The JSON Object is iterated in lexicographical order, which may be different from the inner deserializer order of the tuple

Reply:
Ok ,so we can only accept array value now.